pkg/logbridge/logbridge.go

```diff
@@ -0,0 +1,88 @@
+package logbridge
+
+import (
+	"bufio"
+	"io"
+
+	"github.com/square/p2/pkg/logging"
+)
+
+// This type is agnostic of pipes. os.File conforms to io.Reader and io.Writer
+// so it should not be difficult to migrate
+type Bridge struct {
+	src  io.Reader
+	dest io.Writer
+
+	lines         chan []byte
+	needsTeardown bool
+}
+
+// New takes two files src and dest and returns a composite object of the
+// two.
+func New(src io.Reader, dest io.Writer, capacity int) Bridge {
+	if src == nil {
+		panic("Argument Error: Reader cannot be nil")
+	}
+	if dest == nil {
+		panic("Argument Error: Reader cannot be nil")
+	}
+
+	return Bridge{
+		src:           src,
+		dest:          dest,
+		lines:         make(chan []byte, capacity),
+		needsTeardown: true,
+	}
+}
+
+// Run is responsible for reading from src and writing to dst until src is
+// consumed or an error is encountered. The caller is responsible for closing
+// and flushing the reader and writer.
+func (b *Bridge) Run() error {
```

Comment:
Another thing to note is that this API provides no means of termination (eg quit channel), but since it'll probably be at the level of main, maybe that's not a big deal.